src/test/java/org/opensearch/sdk/TestNetty4Transport.java

```diff
@@ -25,11 +25,14 @@
 public class TestNetty4Transport extends OpenSearchTestCase {
 
     private ThreadPool threadPool;
-    private NettyTransport nettyTransport = new NettyTransport();
+    private ExtensionsRunner extensionsRunner;
+    private NettyTransport nettyTransport;
 
     @BeforeEach
     public void setUp() throws IOException {
         this.threadPool = new TestThreadPool("test");
+        this.extensionsRunner = new ExtensionsRunner();
```

Comment:
(Not for this PR) I wonder if there's value in creating a "Test" version of the ExtensionsRunner.  The no-arg constructor here is used exclusively for testing and I'd like to see it removed and replaced with some TestExtensionsRunner class in the test suite.

Reply:
Created #171 for this.